Yoshi/QAKit/EnvironmentMenu/YoshiEnvironmentManager.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  YoshiEnvironmentManager.swift
+//  Yoshi
+//
+//  Created by Kanglei Fang on 28/08/2017.
+//  Copyright © 2017 Prolific Interactive. All rights reserved.
+//
+
+import Foundation
+
+/// Environement manager contains current and all available environemnts info.
+public protocol YoshiEnvironmentManager: class {
+
+    /// The current HTTP environment.
+    var currentEnvironment: YoshiEnvironment { get set }
```

Comment:
my preference would be go have `currentEnvironment` be `read-only` then have a function like `func updateEnvironment(_ newEnvironment: YoshiEnvironment)` - i find that a bit more explicit